frontend/src/components/common/focusTrap/FocusTrap.tsx

```diff
@@ -0,0 +1,95 @@
+import { Children, cloneElement, useEffect, useRef } from "react";
+
+interface FocusTrapProps extends React.HTMLAttributes<HTMLDivElement> {
+  children: React.ReactElement;
+  onEscapeFocusTrap: () => void;
+}
+
+const getFocusableElements = (
+  element: HTMLElement | ChildNode | null,
+  result: HTMLElement[] = [],
+) => {
+  if (!element || !element.childNodes) return result;
+
+  for (const childNode of element.childNodes) {
+    const childElement = childNode as HTMLElement;
+    if (childElement.tabIndex >= 0) {
+      result.push(childElement);
+    }
+    getFocusableElements(childElement, result);
+  }
+
+  return result;
+};
+
+const FocusTrap = (props: FocusTrapProps) => {
+  const { children, onEscapeFocusTrap, ...others } = props;
+  const child = Children.only(children);
+
+  const focusTrapRef = useRef<HTMLDivElement>(null);
+  const focusableElements = useRef<(HTMLElement | null)[]>([]);
+  const currentFocusIndex = useRef(-1);
+
+  const Compo = cloneElement(child, {
+    ...{ ...others, ...child?.props },
+    tabIndex: -1,
+    ref: focusTrapRef,
+  });
+
+  const focusNextElement = () => {
+    currentFocusIndex.current = (currentFocusIndex.current + 1) % focusableElements.current.length;
+    focusableElements.current[currentFocusIndex.current]?.focus();
+  };
+
+  const focusPrevElement = () => {
+    currentFocusIndex.current =
+      currentFocusIndex.current === 0
+        ? focusableElements.current.length - 1
+        : currentFocusIndex.current - 1;
+    focusableElements.current[currentFocusIndex.current]?.focus();
+  };
```

Comment:
`focusPrevElement`도 `focusNextElement` 함수와 같은 방식으로 동작하도록 하면 좋을 것 같아요!
아래처럼 수정하면 항상 length만큼 더해지니까 음수 인덱스가 나올 일이 없을 것 같은데 이 방식은 어떤가요??

```suggestion
const focusPrevElement = () => {
  currentFocusIndex.current = (currentFocusIndex.current - 1 + focusableElements.current.length) % focusableElements.current.length;
  focusableElements.current[currentFocusIndex.current]?.focus();
};
```

Reply:
요게 더 보기 좋은거 같아서 반영할게요!